File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
from collections import deque
def solution(n, words):
    answer = []
    idx = 0
    count = 1
    queue = []
    queue.append(words[0])
    for i in range(1,len(words)):
        #queue에 마지막에 있는 -1의값이랑 words[i]의 마지막글자가 false고, words[i]가 queue에 있다면
        if not words[i].startswith(queue[-1][-1]) or words[i] in queue:
            answer.append((i%n)+1)
            answer.append((i//n)+1)
            break
        else:
            queue.append(words[i])
    #answer가 비어있으면 즉 끝말잇기가 끝까지 진행됐다면
    if not answer:
        answer.append(0)
        answer.append(0)
    return answer

File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;
public class Solution {
    public int[] solution(int n, String[] words){
        int[] answer = {0,0};
        int cnt = 0;
        Deque<String> wordsGroups = new ArrayDeque<>();
        for(int i=0; i<words.length; i++){
            String currentword = words[i];

            if(wordsGroups.isEmpty()){
                wordsGroups.add(currentword);
            }
            else if(wordsGroups.contains(currentword)){
                answer[0] = i%n +1;
                answer[1] = i/n +1;
          
                break;
            }
            else{
                int wordSize = wordsGroups.peekLast().length()-1;
//                System.out.println(wordsGroups.peekLast().charAt(wordSize));
//                System.out.println(currentword.charAt(0));
                if(wordsGroups.peekLast().charAt(wordSize) == currentword.charAt(0)){
                    wordsGroups.add(currentword);
                }
                else{
                    answer[0] = i%n +1;
                    answer[1] = i/n +1;
                    break;
                }
            }

        }
        return answer;
    }

}